docs/src/pages/customization/styled/styled.md

```diff
@@ -0,0 +1,80 @@
+# `experimentalStyled()`
+
+<p class="description">Utility for creating styled components, compatible with the Material-UI's core components.</p>
+
+## Introduction
+
+All Material-UI components use the `experimentalStyled()` utility from `@material-ui/core/styles` under the hood.
+The utility is built on top of [emotion's `styled()` utility](https://emotion.sh/docs/styled) but provides additional options that can be used for adding some features that are specific to the Material-UI components.
+The additional features include:
+
+- support for [`theme.components[name].styleOverrides`](/customization/theme-components/#global-style-overrides)
+- support for [`theme.components[name].variants`](/customization/theme-components/#adding-new-component-variants)
+- support for [the `sx` prop](/system/basics/#the-sx-prop)
+
+### What problems does it solve?
+
+The utility can be used as a replacement for emotion's/styled-components' styled() utility.
+It aims to solve the same problem, but also provides the following benefits:
+
+1. It uses the default Material-UI's `theme` if no theme is available in the context
+2. It supports the theme's `styleOverrides` and `variants` to be applied, based on the `name` applied in the options (can be skipped)
+3. It adds support for the `sx` prop automatically (can be skipped)
+4. It adds by default `shouldForwardProp` option that is taking into account all props used internally in the Material-UI components (can be overridden)
+
+## API
+
+### `experimentalStyled(Component, [options], [muiOptions])(styles) => Component`
+
+#### Arguments
+
+1. `Component`: The component that will be wrapped.
+2. `options` (_object_ [optional]):
+
+   - `options.shouldForwardProp` (_`(props: string) => boolean`_ [optional]): Indicates whether the `prop` should be forwared to the `Component`.
+   - `options.label` (_string_ [optional]): The suffix of the style sheet. Useful for debugging.
+   - The other keys are forwarded to the options argument of emotion's [styled([Component], [options])](https://emotion.sh/docs/styled).
+
+3. `muiOptions` (_object_ [optional]):
+
+   - `muiOptions.name` (_string_ [optional]): The key used under `theme.components` for specifying `styleOverrides` and `variants`. Used also for generating the `label`.
+   - `muiOptions.slot` (_string_ [optional]): If `Root`, it automatically applies the theme's `styleOverrides` & `variants`
+   - `muiOptions.overridesResolver` (_(props: object, styles: Record<string, styles>) => styles_ [optional]): Function that returns styles based on the props and the `theme.components[name]styleOverrides` object.
+   - `muiOptions.skipVariantsResolver` (_boolean_): Disables the automatic resolver for the `theme.components[name].variants`
+   - `muiOptions.skipSx` (_bool_ [optional]): Disables the `sx` prop on the component.
```

Comment:
How is having two option bags relevant to the usage?

Reply:
Should we have a single `options` argument instead of two?

Reply:
We could merge the options. Should I create a PR before proceeding with this?

Reply:
Yeah, this is more of a general question now that I have the full interface+docs in front of me. So it seems like we can merge these two bags. Unless there's a meaningful difference for users for an option being a "normal" option or a "mui" option. But I suspect you wouldn't really care about whether an option is in `options` or `muiOptions`.

Reply:
Semantically the first option is emotion's original optional, the second are the mui specific. I don't think people will care about this, it may even be more confusing as to why we have two options, for people not aware too much with the specifics of emotion. Let me merge the options and update this PR.